pkg/plugins/generic/generic_plugin.go

```diff
@@ -83,7 +83,7 @@ func (p *GenericPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetworkN
 	err = nil
 	p.DesireState = new
 
-	needDrain = needDrainNode(new.Spec.Interfaces, old.Status.Interfaces)
+	needDrain = needDrainNode(new.Spec.Interfaces, new.Status.Interfaces)
```

Comment:
This is going to work because the interfaces in the status part are updated asynchronously, right? And in any case, not in the version we are handling now.

Reply:
The status is not always updated asynchronously, as in this patch I add a `syncCh` channel to block the daemon until writer updates the status. This is to make sure in each update handling, the config daemon can get the latest status which is modified by last round.

Reply:
Ah, I missed that! Thanks!